core/src/main/scala/com/softwaremill/react/kafka2/Consumer.scala

```diff
@@ -75,7 +89,7 @@ object ManualCommitConsumer {
   }
 }
 
-class ManualCommitConsumer[K, V](consumerProvider: ConsumerProvider[K, V])
+class ManualCommitConsumer[K, V](consumerProvider: () => KafkaConsumer[K, V])
```

Comment:
That was my initial idea: having composable immutable reusable items like `KafkaConsumer => KafkaConsumer` which perform some setup for consumer. 

The problem with this approach is creating initial `KafkaConsumer`, because I want to force some properties will be set (like servers list, serializers and deserializers).

I modeled this with `ConsumerProvider` which should provide type safe API to configure kafka properties (not done yet, but can be implemented). First implementation takes `ConsumerProvider` as parameter, but next I ask a question: should we keep coupling with graph API and consumer setup API at all? And the answer is no, because no coupling is better then coupling :)

Hope I explained all my decision path well.